source/change-streams/tests/change-streams.yml

```diff
@@ -0,0 +1,197 @@
+collection_name: &collection_name "test"
+database_name: &database_name "change-stream-tests"
+collection_name_2: &collection_name_2 "test2"
+database_name_2: &database_name_2 "change-stream-tests-2"
+defaults:
+  minServerVersion: "3.6.0"
+  topology: replicaset
+  target: collection
+  numChanges: 1
+  result: {}
+  expectations: []
+  operations:
+    -
+      database: *database_name
+      collection: *collection_name_2
+      commandName: insert
+      arguments:
+        - { x: 1 }
+    -
+      database: *database_name_2
+      collection: *collection_name
+      commandName: insert
+      arguments:
+        - { y: 2 }
+    -
+      database: *database_name
+      collection: *collection_name
+      commandName: insert
+      arguments:
+        - { z: 3 }
+tests:
+  -
+    description: "`$changeStream` must be the first stage in a change stream pipeline sent to the server"
+    expectations:
+      - 
+        command_started_event:
+          command:
+            aggregate: *collection_name
+            pipeline:
+              - {$changeStream: {}}
+          command_name: aggregate
+          database_name: *database_name
+  -
+    description: The watch helper must not throw a custom exception when executed against a single server topology, but instead depend on a server error
+    topology: single
+    operations: []
+    result:
+      error:
+        errmsg: The $changeStream stage is only supported on replica sets
+        code: 40573
+  -
+    description: ChangeStream must continuously track the last seen resumeToken
+    skip: true
+  -
+    description: ChangeStream will throw an exception if the server response is missing the resume token
+    skip: true
+  -
+    description: ChangeStream will automatically resume one time on a resumable error (including not master) with the initial pipeline and options, except for the addition/update of a resumeToken.
+    skip: true
+  -
+    description: ChangeStream will not attempt to resume on a server error
+    skip: true
+  -
+    description: ChangeStream will perform server selection before attempting to resume, using initial readPreference
+    skip: true
+  -
+    description: Ensure that a cursor returned from an aggregate command with a cursor id, and an initial empty batch, is not closed on the driver side.
```

Comment:
Does this require a failpoint?

Reply:
No, this was another one that I was not sure how to express as a test.

Reply:
Seems more confusing to have these placeholders in here that we don't know how to implement. Consider taking them out and opening a SPEC ticket instead to track the desire.